languages/csharp/csharp.go

```diff
@@ -1,29 +1,57 @@
 package csharp
 
 import (
-	"log"
-
 	sitter "github.com/smacker/go-tree-sitter"
-	"github.com/smacker/go-tree-sitter/java" // FIXME: C#
+	"github.com/smacker/go-tree-sitter/csharp"
 	"github.com/src-d/imports/languages/tsitter"
 )
 
+const (
+	query = `
+		(using_directive (identifier_name) @name)
+		(using_directive (qualified_name) @name)
+	`
+)
+
 func init() {
 	tsitter.RegisterLanguage(language{})
 }
 
-type language struct{}
+type language struct {
+	q *sitter.Query
+}
 
 func (language) Aliases() []string {
 	return []string{"C#"}
 }
 
 func (l language) GetLanguage() *sitter.Language {
-	return java.GetLanguage()
+	return csharp.GetLanguage()
 }
 
 func (l language) Imports(content []byte, n *sitter.Node) ([]string, error) {
-	log.Println(n)
+	if l.q == nil {
```

Comment:
This should be a local variable. You are currently not reusing it since the method is not on a pointer receiver.
If it can be reused that way though, I would suggest adding a global variable and initialize it in the `init` instead. Or use `sync.Once` to populate it only when `Imports` is called.

Reply:
I made global var and initialized by once. I don't want to compile the pattern for every request.